### PR TITLE
Use the name from the property in exceptions

### PR DESCRIPTION
**Pull request by gskierk**

### CODE REVIEW

**Comment by PowerKiKi** [comment]
What is your use case to have a different name ? How do you use it ?

**Comment by gskierk** [comment]
I may want to rename this scalar to anything else... Current implementation allows to rename it (constructor, public property) anyway, but then the messages become inconsistent. So, we could attach the name to exception messages (this PR) or disable the possibility to rename it.

File: src/UploadType.php [hunk 1]
```diff
@@ -22,16 +22,21 @@ class UploadType extends ScalarType
...
-    public $description
-        = 'The `Upload` special type represents a file to be uploaded in the same HTTP request as specified by
- [graphql-multipart-request-spec](https://github.com/jaydenseric/graphql-multipart-request-spec).';
+    public $description;
+
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        $this->description = 'The `' . $this->name . '` special type represents a file to be uploaded in the same HTTP request as specified by [graphql-multipart-request-spec](https://github.com/jaydenseric/graphql-multipart-request-spec).';
```
Comment (PowerKiKi) [inline src/UploadType.php:31]: You are overwriting the potentially custom description given in `$config`. So by trying to allow customizing the name, you don't allow to customize the description anymore. That should not be the case.

**Comment by PowerKiKi** [comment]
It was never meant to be possible to rename the type. However you are correct that it technically is possible to do so, either by constructor, or by public property.

If anything, I'd rather prevent customizing. Unfortunately we cannot eliminate the public property, since it is a requirement of `webonyx/graphql-php`. So even if we "protect" the constructor, we still don't have a solution for the property.

Honestly I don't think it's worth the effort to do anything at all. I cannot imagine a use-case where we really *must* rename the type. And "Upload" should be good enough for the vast majority of use-cases.

**Pull request closed without merging**